File: java/calendar/src/test/java/com/rbleggi/calendar/CalendarRemoveTest.java
package com.rbleggi.calendar;

import org.junit.jupiter.api.Test;
import java.time.LocalDateTime;
import java.util.Set;
import static org.junit.jupiter.api.Assertions.*;

class CalendarRemoveTest {
    @Test
    void removeMeetingReturnsTrue() {
        var calendar = new Calendar();
        var user = new User("1", "Alice");
        var meeting = new Meeting("m1", "Sync", LocalDateTime.of(2025, 1, 1, 9, 0), LocalDateTime.of(2025, 1, 1, 10, 0), Set.of(user));
        calendar.bookMeeting(meeting);
        assertTrue(calendar.removeMeeting("m1"));
    }

    @Test
    void removeMeetingReturnsFalseForMissing() {
        var calendar = new Calendar();
        assertFalse(calendar.removeMeeting("nonexistent"));
    }
}

File: kotlin/calendar/src/test/kotlin/com/rbleggi/calendar/CalendarRemoveTest.kt
package com.rbleggi.calendar

import java.time.LocalDateTime
import kotlin.test.Test
import kotlin.test.assertTrue
import kotlin.test.assertFalse

class CalendarRemoveTest {
    @Test
    fun `removeMeeting returns true for existing`() {
        val calendar = Calendar()
        val user = User("1", "Alice")
        val meeting = Meeting(title = "Test", start = LocalDateTime.of(2025, 1, 1, 9, 0), end = LocalDateTime.of(2025, 1, 1, 10, 0), attendees = setOf(user))
        calendar.bookMeeting(meeting)
        assertTrue(calendar.removeMeeting(meeting.id))
    }

    @Test
    fun `removeMeeting returns false for missing`() {
        val calendar = Calendar()
        assertFalse(calendar.removeMeeting("nonexistent"))
    }
}
